### PR TITLE
Add shell verifier

### PR DESCRIPTION
**Pull request by sawanoboly**
Hi, I've created shell verifier. It's useful for run [infrataster](http://infrataster.net) by kitchen.

Features:

- exec shell command locally by shellout.
- fail when its exit_code `!==0`
- pass state via env, such as KITCHEN_HOSTNAME, KITCHEN_PORT, etc...
- option `:remote_exec`: run command on instance.

Configration example

```
platforms:
  - name: ubuntu-12.04
  - name: ubuntu-14.04

verifier:
  name: shell
  command: bundle exec rake
  sleep: 5
```

And sample rake

```
task :default do
  puts "Hello #{ENV['KITCHEN_HOSTNAME']}:#{ENV['KITCHEN_PORT']}"
end
```

Results

```
$ kitchen verify 
-----> Starting Kitchen (v1.4.1.dev)
-----> Verifying <default-ubuntu-1204>...
       [Shell] Verify on instance=#<Kitchen::Instance:0x007fd80e111ac8> with state={:hostname=>"127.0.0.1", :port=>"2200", :username=>"vagrant", :ssh_key=>"...", :last_action=>"verify"}
.....
Hello 127.0.0.1:2200
       Finished verifying <default-ubuntu-1204> (0m5.66s).
-----> Verifying <default-ubuntu-1404>...
       [Shell] Verify on instance=#<Kitchen::Instance:0x007fd80c41c148> with state={:hostname=>"127.0.0.1", :port=>"2222", :username=>"vagrant", :ssh_key=>"...", :last_action=>"verify"}
.....
Hello 127.0.0.1:2222
       Finished verifying <default-ubuntu-1404> (0m5.67s).
-----> Kitchen is finished. (0m11.59s)
```

### CODE REVIEW

**Comment by mizzy** [comment]
:+1:

**Comment by ryotarai** [comment]
+1

**Comment by cl-lab-k** [comment]
:+1:

**Comment by marcy-terui** [comment]
:+1:

**Comment by tkak** [comment]
+1

**Comment by sawanoboly** [comment]
Any updates? @test-kitchen/owners

**Comment by sawanoboly** [comment]
Added: Support environment variables to pass to shell following.

```
KITCHEN_INSTANCE=default-ubuntu-1404
KITCHEN_SUITE=default
KITCHEN_PLATFORM=ubuntu-14.04
```

**Comment by ytsarev** [comment]
well, i would expect shell verifier to run shell commands on instance, not locally.

**Comment by ytsarev** [comment]
@sawanoboly I've sent you PR to gem repo https://github.com/higanworks/kitchen-verifier-shell/pull/1

**Comment by sawanoboly** [comment]
@ytsarev 

When I would like to run shell on remote instance to verify, I'll use `kitchen exec`.

Case: single instance

```
verifier:
  name: shell
  command: kitchen exec -c command( or path_to_test.sh)
```

Case: multi instance

```
verifier:
  name: shell
  command: ./path_to_test.sh
```

in test.sh

```
kitchen exec $KITCHEN_INSTANCE -c command( or path_to_test.sh)
```

Some awesome test-suites have the similar advantages  (such as Serverspec, Infrataster).

They can test server instances via remote, we don't need to install any agent softwares.
So, I'd like to follow the how to use they prepared just as it is.

**Comment by ytsarev** [comment]
@sawanoboly well, serverspec is exactly my use case. But in this testing scenario i would like to run it isolated within an instance under test

**Comment by sawanoboly** [comment]
update: add `:remote_exec` option.

cherry-picked from https://github.com/higanworks/kitchen-verifier-shell/pull/2

**Comment by sawanoboly** [comment]
@fnichol @tyler-ball  Could you please review ?

File: lib/kitchen/verifier/shell.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      default_config :sleep, 0
+      default_config :command, "true"
+      default_config :shellout_opts, {}
+      default_config :live_stream, $stdout
+      default_config :remote_exec, false
+
+      # (see Base#call)
+      def call(state)
+        info("[#{name}] Verify on instance=#{instance} with state=#{state}")
+        sleep_if_set
```
Comment (tyler-ball) [inline lib/kitchen/verifier/shell.rb]: What is the use case for having a configurable sleep?

Comment (sawanoboly) [inline lib/kitchen/verifier/shell.rb]: Some services need initializing itself when first started. Such as jenkins, pentaho-ce and etc. I want to wait few seconds before start test.

**Comment by cl-lab-k** [comment]
Hi, any updates? This feature is very useful. I hope it is merged :smile:

**Comment by sawanoboly** [comment]
@fnichol @tyler-ball Any updates?

**Comment by schisamo** [comment]
:+1: This looks great! I would squash things down into a single commit

**Comment by tyler-ball** [comment]
Squashed and merged in https://github.com/test-kitchen/test-kitchen/commit/cc99dc5e17a2e2b84c24d53fcbca463f8a8f45dd - thanks @sawanoboly !

**Comment by thedrow** [comment]
Where can I find documentation about the shell verifier?

**Comment by sawanoboly** [comment]
Hi, @thedrow 

I wrote all usage to head of this issue.

**Comment by thedrow** [comment]
It'd be nice to have this included in test kitchen's documentation. I shell open a new issue about it.

**Pull request closed without merging**